apps/ledger-live-mobile/src/components/SendRowsFee.tsx

```diff
@@ -37,9 +37,29 @@ export default ({
   ...props
 }: Props) => {
   const mainAccount = getMainAccount(account, parentAccount);
+
   // eslint-disable-next-line no-prototype-builtins
   if (perFamily.hasOwnProperty(mainAccount.currency.family)) {
     const C = perFamily[mainAccount.currency.family as keyof typeof perFamily];
+
+    /**
+     * Evm familly type does not have networkInfo property but we still want to
+     * display the fee selection logic
+     */
+    if (mainAccount.currency.family === "evm") {
```

Comment:
Isn't there a better way to do it ? Shouldn't adding a SendRowFee component in evm family be enough ? Looks like it should be in perFamily map

Reply:
> Isn't there a better way to do it ?

Probably, but not that I am aware of

> Shouldn't adding a SendRowFee component in evm family be enough ?

We already have a `SendRowFee` component in the evm family folder (as well as in other families folders). But this component here seems to be the "root" component that will display the correct family specific component from the `perFamily` generated file based on the account family.

> why has it been done like this for bitcoin ?

I don't know. But it seems to be legacy code and might not be what we want.

```ts
  if (perFamily.hasOwnProperty(mainAccount.currency.family)) {
    const C = perFamily[mainAccount.currency.family as keyof typeof perFamily];
    // FIXME: looks like a hack, need to find how to handle networkInfo properly
    return (transaction as BitcoinTransaction)?.networkInfo ? (
      <C
        {...props}
        setTransaction={setTransaction}
        transaction={transaction}
        account={account}
        parentAccount={parentAccount}
        navigation={navigation}
        route={route}
      />
    ) : null;
  }
``` 

Indeed, here we will only display the `SendRowsFee` component for families that have such component defined in their own specific family folder (this is logic) **AND** that have a `networkInfo` property in their transaction.
This second condition seems odd because not all families have a `networkInfo` property in their transaction. For example `algorand`, but also the new `evm`. But these families have their own `SendRowsFee` component defined in their family folder. Same thing for `celo`, `polkadot`, `solana` (here we have an hardcoded patch, cf. [this](https://github.com/LedgerHQ/ledger-live/blob/develop/libs/ledger-live-common/src/families/solana/js-prepareTransaction.ts#L113-L114)).
So, based on this logic, as of today the `SendRowsFee` for `algorand`, `celo` and `polkadot` would not be displayed.

I would rather get rid of the second condition altogether to have something like that.
Then each specific coin has the responsibility, if need be, to check that they have all the info needed to display their own `SendRowsFee` component, and return `null` otherwise (which is what is done in the `evm` one for example, cf. [here](https://github.com/LedgerHQ/ledger-live/pull/3945/files#diff-06721b193f06f6bec81920428dbc92a0a4778dae78488729526832544a038d01R91-R93)) 

```ts
  if (perFamily.hasOwnProperty(mainAccount.currency.family)) {
    const C = perFamily[mainAccount.currency.family as keyof typeof perFamily];
    return (
      <C
        {...props}
        setTransaction={setTransaction}
        transaction={transaction}
        account={account}
        parentAccount={parentAccount}
        navigation={navigation}
        route={route}
      />
    );
  }
```  

But I think this (little) rework should not be handled in this PR but rather in it's own PR to avoid unexpected side effects not related with the "create evm send flow" topic.

cc. @haammar-ledger FYI
cc. @gre maybe you have some clues on where this `(transaction as BitcoinTransaction)?.networkInfo` condition comes from? From the history, I see it is quite old (before the monorepo), hence my guess on it's possible legacy status.

Reply:
FYI, proposed solution implemented in this PR https://github.com/LedgerHQ/ledger-live/pull/4010

Reply:
PR #4010 looks good ! Let's just check if there is no regression especially on btc